apm-agent-api/src/main/java/co/elastic/apm/api/Transaction.java

```diff
@@ -229,6 +229,15 @@
      */
     Transaction setUser(String id, String email, String username, String domain);
 
+    /**
+     * Provides a way to manually set the {@code http.request.socket.remote_address} field.
```

Comment:
Maybe `http.request.socket.remote_address` should be `client.source.ip` (see elastic/apm-server#6152)?